src/components/layout/Header.tsx

```diff
@@ -45,6 +48,30 @@ const Header = () => {
 
   const [mlMenu, setMlMenu] = useState<(EventTarget & HTMLElement) | null>(null);
   const [docsMenu, setDocsMenu] = useState<(EventTarget & HTMLElement) | null>(null);
+  const [loginMenu, setLoginMenu] = useState<(EventTarget & HTMLElement) | null>(null);
+
+  const auth = useAuth();
+
+  const loginLink = auth.isAuthenticated ? (
```

Comment:
Little nitpicky but I think this is easier to read if you place it directly in JSX instead of here. As a bonus, you only have to conditionally render the text instead of the entire component.